firmware/Core/Src/littlefs/littlefs_benchmark.c

```diff
@@ -0,0 +1,153 @@
+
+#include "main.h"
+#include "littlefs/littlefs_benchmark.h"
+#include "littlefs/littlefs_helper.h"
+
+#include <string.h>
+#include <stdio.h>
+
+/// @brief Benchmarks the write/read operations on the LittleFS file system.
+/// @details This function will write test data to a static filename, read it back, and verify the read data.
+/// @param write_chunk_size Number of bytes to write in each chunk.
+/// @param write_chunk_count Number of chunks to write.
+/// @param response_str 
+/// @param response_str_len 
+/// @return 0 on success. >0 if there was an error.
+uint8_t LFS_benchmark_write_read(uint16_t write_chunk_size, uint16_t write_chunk_count, char* response_str, uint16_t response_str_len) {
+    const char file_name[] = "benchmark_test.txt";
+    response_str[0] = '\0';
+
+    uint8_t expected_checksum = 0;
+
+    // Prep some data to write
+    uint8_t write_buffer[write_chunk_size];
+    for (uint32_t i = 0; i < write_chunk_size; i++) {
+        const uint8_t b = (i + 42) % 256;
+        write_buffer[i] = b;
+    }
+
+    // Open file for writing
+    const uint32_t write_open_start_time = HAL_GetTick();
+    lfs_file_t file;
+    const int8_t open_result = lfs_file_opencfg(&LFS_filesystem, &file, file_name, LFS_O_WRONLY | LFS_O_CREAT | LFS_O_TRUNC, &LFS_file_cfg);
```

Comment:
Why not `lfs_file_open` instead of `lfs_file_opencfg`?

Reply:
`lfs_file_opencfg` uses a statically-allocated "file buffer" (in the `LFS_file_cfg` struct), which is required to prevent LFS from calling `malloc`. That change is a non-benchmarking change in this PR (which maybe should have been separated out into a separate PR, ideally).